src/message_interface/message_type.rs

```diff
@@ -117,6 +121,54 @@ pub enum MessageType {
     EmitTestEvent(WalletEvent),
 }
 
+// Custom Debug implementation to not log secrets
+impl Debug for MessageType {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result {
+        match self {
+            MessageType::CreateAccount(account) => write!(f, "CreateAccount({:?})", account),
+            MessageType::GetAccount(identifier) => write!(f, "GetAccount({:?})", identifier),
+            MessageType::GetAccounts => write!(f, "GetAccounts"),
+            MessageType::CallAccountMethod { account_id, method } => write!(
+                f,
+                "CallAccountMethod{{ account_id: {:?}, method: {:?} }})",
```

Comment:
A trailing bracket here; also is it the convention here to use two curly brackets?

Reply:
With a single `{` one gets
```
error: invalid format string: expected `'}'`, found `'a'`
   --> src/message_interface/message_type.rs:133:37
    |
133 |                 "CallAccountMethod{ account_id: {:?}, method: {:?} }",
    |                                   - ^ expected `}` in format string
    |                                   |
    |                                   because of this opening brace
    |
    = note: if you intended to print `{`, you can escape it using `{{`
    ```